aptos-move/aptos-vm/src/lib.rs

```diff
@@ -153,14 +153,15 @@ pub trait VMValidator {
 }
 
 /// This trait describes the VM's execution interface.
-pub trait VMExecutor: Send + Sync {
-    // NOTE: At the moment there are no persistent caches that live past the end of a block (that's
-    // why execute_block doesn't take &self.)
-    // There are some cache invalidation issues around transactions publishing code that need to be
-    // sorted out before that's possible.
+pub trait VMBlockExecutor: Send + Sync {
```

Comment:
Can we move this trait outside of `aptos-vm`? E.g., `aptos-types` or something? WIth your changes does not belong here anymore I think

Reply:
Also, do we want the executor to be `Send + Sync`? I am not sure, unless its stateless, otherwise we can end up like MoveVM where it was thread-safe but used for speculation.

Reply:
Regarding moving out of this crate: a TODO suffices, if this adds more work

Reply:
Idea is that one of these is created for the whole execution , and potentially across blocks. So Send+Sync is needed.

i.e. for native executors, we have x-block cache of values there, for example. 
You could put the x-block module cache there - so that methods below don't need to have it as an argument. you would call your "before execution" and "after execution" calls inside of the VMBlockExecutor implementation. Though not sure if that would work correctly for your case

also we would need to make sure execution across blocks in production reuses the instance (I made sure that is the case for executor benchmark, as I am using this only for non-prod executors thus far)